doc/source/container_builder.rst

```diff
@@ -0,0 +1,282 @@
+
+Requirements
+------------
+
+The following components needs to be installed on the build system:
+
+* Dice - a containment build system for KIWI.
+
+* Docker - a container framework based on the Linux
+  container support in the kernel.
+
+* Docker Build Container for KIWI.
+
+* optionally Vagrant - a framework to run, provision and control
+  virtual machines and container instances. Vagrant has a very nice
+  interface to provision a machine prior to running the actual build.
+  It also supports docker as a provider which makes it a perfect fit
+  for complex provisioning tasks in combination with the Docker
+  container system.
+
+Installing and Setting up Dice
+------------------------------
+
+The Dice packages and sources are available at the following locations:
+
+* Build service project: https://build.opensuse.org/project/show/Virtualization:Appliances:ContainerBuilder
+* Sources: https://github.com/SUSE/dice
+
+.. code:: bash
+
+    $ sudo zypper in ruby2.1-rubygem-dice
+
+Installing and Setting up Docker
+--------------------------------
+
+Docker packages are usually available with the used distribution.
+
+.. code:: bash
+
+    $ sudo zypper in docker
+
+Make sure that the user, who is intended to build images, is a member
+of the `docker` group. Run the following command:
+
+.. code:: bash
+
+    $ sudo useradd -G docker <builduser>
+
+It is required to logout and login again to let this change
+become active.
+
+Once this is done you need to setup the Docker storage backend.
+By default Docker uses the device mapper to manage the storage for
+the containers it starts. Unfortunately, this does not work if the
+container is supposed to build images because it runs into conflicts
+with tools like :command:`kpartx` which itself maps devices using
+the device mapper.
+
+Fortunately, there is a solution for Docker which allows us to use
+Btrfs as the storage backend. The following is only required if your
+host system root filesystem is not btrfs:
+
+.. code:: bash
+
+    $ sudo qemu-img create /var/lib/docker-storage.btrfs 20g
+    $ sudo mkfs.btrfs /var/lib/docker-storage.btrfs
+    $ sudo mkdir -p /var/lib/docker
+    $ sudo mount /var/lib/docker-storage.btrfs /var/lib/docker
+
+    $ sudo vi /etc/fstab
+
+      /var/lib/docker-storage.btrfs /var/lib/docker btrfs defaults 0 0
+
+    $ sudo vi /etc/sysconfig/docker
+
+      DOCKER_OPTS="-s btrfs"
+
+Finally start the docker service:
+
+.. code:: bash
+
+    $ sudo systemctl restart docker
+
+Installing and Setting up the Build Container
+----------------------------------------------
+
+In order to build in a contained environment Docker has to start a
+privileged system container. Such a container must be imported before
+Docker can use it. The Build Container is provided to you as a
+service and build with KIWI in the project
+at https://build.opensuse.org/project/show/Virtualization:Appliances:Images
+On a regular basis the result image is pushed
```

Comment:
What does "On a regular basis" actually means? Is there also an irregular basis? ;-)

Maybe simplify by just writing: "The result image is pushed to ..."